pkg/controller/handler.go

```diff
@@ -12,41 +12,85 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 )
 
+const SAS_PREFIX = "k8scc01covidacr.azurecr.io/sas:"
+const FEATURES_LABEL = "state.aaw.statcan.gc.ca/employee-only-features"
+const RB_LABEL = "state.aaw.statcan.gc.ca/non-employee-user"
+
 func sasImage(pod *corev1.Pod) bool {
-	image := pod.Spec.Containers[0].Image
-	sasImage := strings.HasPrefix(image, "k8scc01covidacr.azurecr.io/sas:")
-	return sasImage
+
+	for _, container := range pod.Spec.Containers {
+		sasImage := strings.HasPrefix(container.Image, SAS_PREFIX)
+		if sasImage {
+			return true
+		}
+	}
+
+	return false
 }
 
-func (c *Controller) handleProfile(profile *v1.Profile) error {
-	ctx := context.Background()
+func isEmployee(email string) bool {
+
+	employeeDomains := [2]string{"cloud.statcan.ca", "statcan.gc.ca"} // Intialized with values
+	for _, domain := range employeeDomains {
+		if strings.HasSuffix(email, domain) {
+			return true
+		}
+	}
+	return false
+}
 
+func (c *Controller) handleProfile(profile *v1.Profile) error {
 	namespace := profile.Name
 	hasEmpOnlyFeatures := false
 
+	// label set from rolebindings, if there are no rolebindings, then the default is that they are external
+	nonEmployeeUser := true
+
 	pods, err := c.podLister.Pods(namespace).List(labels.Everything())
+
+	if err != nil {
+		return err
+	}
+
+	roleBindings, err := c.roleBindingLister.RoleBindings(namespace).List(labels.Everything())
+
 	if err != nil {
 		return err
 	}
+
 	for _, pod := range pods {
 		if sasImage(pod) {
 			hasEmpOnlyFeatures = true
 			break
 		}
 	}
 
+	for _, roleBindings := range roleBindings {
+		for _, subject := range roleBindings.Subjects {
+			email := subject.Name
+			if strings.Contains(email, "@") {
+				if isEmployee(email) {
+					nonEmployeeUser = false
+					break
```

Comment:
Might be an opportunity to encapsulate the inner loop in a separate function.

Could write something similar to what you did here https://github.com/StatCan/aaw-profile-state-controller/pull/7/files#diff-54fe1371798f1e212124dbfe7803367d2bf18dc0738c8ed61719b024c6af2f15L15-R28 but to check if user is an employee or not.